sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/MethodOverloadOptionalParameter.cs

```diff
@@ -147,18 +148,28 @@ private static bool IsMethodHidingOriginal(IMethodSymbol candidateHidingMethod,
         {
             for (int i = 0; i < candidateHidingMethod.Parameters.Length; i++)
             {
-                if (!candidateHidingMethod.Parameters[i].Type.Equals(method.Parameters[i].Type))
+                var candidateParam = candidateHidingMethod.Parameters[i];
+                var methodParam = method.Parameters[i];
+
+                if (!Equals(candidateParam.Type, methodParam.Type) &&
+                    !AreTypeParameters(candidateParam, methodParam) &&
+                    !Equals(candidateParam.Type.OriginalDefinition, methodParam.Type.OriginalDefinition))
                 {
                     return false;
                 }
 
-                if (candidateHidingMethod.Parameters[i].IsOptional != method.Parameters[i].IsOptional)
+                if (candidateParam.IsOptional != methodParam.IsOptional)
                 {
                     return false;
                 }
             }
 
             return true;
         }
+
+        private static bool AreTypeParameters(IParameterSymbol p1, IParameterSymbol p2)

```

Comment:
I would remove this method and put the 2 conditions directly on the if. I find the current method call confusing when reading it.